In [2]:
import os
import json

PATH = "/hpc/projects/group.califano/GLM/scGraphLLM/scripts/slurm_out"

for dr in os.listdir(PATH):
    print(dr)

slurm_out_16726641
.ARACNe3_cached


In [10]:
run = "16726641"

success_file = open(f"/hpc/projects/group.califano/GLM/scGraphLLM/scripts/slurm_out/slurm_out_{run}/check_out/success.txt", "r")
success_file = [dir[:-1] for dir in success_file]

PATH = "/hpc/projects/group.califano/GLM/data/cellxgene/data/complete_data"

cell_count = 0
for cell_type in success_file:
    for f in os.listdir(f"{PATH}/{cell_type}"):
        if "info" in f:
            with open(f'{PATH}/{cell_type}/{f}', 'r') as file:
                data = json.load(file)
        
    cell_count += data["metacells"]["n_cells"]

print(f"{cell_count} cells are in the dataset!")

2149185 cells are in the dataset!


In [10]:
PATH = "/hpc/projects/group.califano/GLM/data/cellxgene/data/complete_data"

failed = []
cell_count = 0
for cell_type in os.listdir(PATH):
    try:
        with open(f'{PATH}/{cell_type}/info.json', 'r') as file:
            data = json.load(file)
            
        cell_count += data['stats']["metacells"]["n_cells"]
    except:
        failed.append(cell_type)
    
print(f"\n{cell_count} cells are in the dataset!")
print(len(failed))


4448154 cells are in the dataset!
43


In [12]:
p = "/hpc/projects/group.califano/GLM/scGraphLLM/scripts/slurm_out/slurm_out_16703767/check_out/failed.txt"
file = open(p, 'r')
file_str = "".join(file)
file.close()

count = 0
for f in failed:
    if f in file_str:
        count += 1

count

37

## Before preprocessing

In [1]:
import os
import sys
import scanpy as sc
import anndata as ad
from tqdm import tqdm
import contextlib

In [18]:
p = "/hpc/projects/group.califano/GLM/data/cellxgene/data/cell_type_all/abnormal_cell/partitions/pan-cancer_0.h5ad"

adata = sc.read_h5ad(p)
adata.X.toarray().shape

(11410, 19221)

In [19]:
p = "/hpc/projects/group.califano/GLM/data/cellxgene/data/cell_type_all"

count = 0
for d in tqdm(os.listdir(p)):
    for h in os.listdir(f'{p}/{d}/partitions'):
        # Look through all the h5ad files and check number of cells (rows in GEMs)
        adata = sc.read_h5ad(f'{p}/{d}/partitions/{h}')
        count += adata.X.shape[0]
        del adata
        
count

  1%|▏         | 9/666 [00:06<08:15,  1.32it/s]


KeyboardInterrupt: 

174,872,658

In [6]:
run = "16703767"

success_file = open(f"/hpc/projects/group.califano/GLM/scGraphLLM/scripts/slurm_out/slurm_out_{run}/check_out/success.txt", "r")
success_file = [dir[:-1] for dir in success_file]

p = "/hpc/projects/group.califano/GLM/data/cellxgene/data/cell_type_all"
count = 0
for d in tqdm(success_file):
    for h in os.listdir(f'{p}/{d}/partitions'):
        # Look through all the h5ad files and check number of cells (rows in GEMs)
        adata = sc.read_h5ad(f'{p}/{d}/partitions/{h}')
        count += adata.X.shape[0]

count

100%|██████████| 261/261 [21:19<00:00,  4.90s/it]


113692215

In [1]:
import os
import shutil
import torch

In [10]:
torch.load(f"/hpc/projects/group.califano/GLM/data/pilotdata_1024/train/aracne_1024_13827.pt")

RuntimeError: PytorchStreamReader failed reading file data/1: invalid header or archive is corrupted

In [2]:
for d in os.listdir("/hpc/projects/group.califano/GLM/data/pilotdata_1024/train/"):
    print(d)
    torch.load(f"/hpc/projects/group.califano/GLM/data/pilotdata_1024/train/{d}")
# torch.load("/hpc/projects/group.califano/GLM/data/pilotdata_debug/train/wababa.pt")

aracne_1024_40634.pt
aracne_1024_33563.pt
aracne_1024_10675.pt
aracne_1024_42753.pt
aracne_1024_36412.pt
aracne_1024_13741.pt
aracne_1024_13827.pt


RuntimeError: PytorchStreamReader failed reading file data/1: invalid header or archive is corrupted

113,692,215

Check number of cells in the successfully loaded cell-types (before processing) - does this match after processing?

## Filter corrupted files in cache

In [1]:
import os
import shutil
import torch

In [ ]:
SOURCE_PATH = "/hpc/projects/group.califano/GLM/data/pilotdata_1024"

os.makedirs(f"{SOURCE_PATH}_filtered", exist_ok=True)
for SET in ["train", "valHG", "valSG"]:
    os.makedirs(f"{SOURCE_PATH}_filtered/{SET}", exist_ok=True)
    for dr in os.listdir(f"{SOURCE_PATH}/{SET}"):
        try:
            torch.load(f"{SOURCE_PATH}/{SET}/{dr}")
        except:
            continue